lib/msal-react/src/utils/utilities.ts

```diff
@@ -22,3 +22,19 @@ export function getChildrenOrFunction<T>(
 type SetDifference<A, B> = A extends B ? never : A;
 type SetComplement<A, A1 extends A> = SetDifference<A, A1>;
 export type Subtract<T extends T1, T1 extends object> = Pick<T,SetComplement<keyof T, keyof T1>>;
+
+/**
+ * Helper function to determine whether 2 arrays are equal
+ * Used to avoid unnecessary state updates
+ * @param arrayA 
+ * @param arrayB 
+ */
+export function arraysAreEqual(arrayA: Array<object>, arrayB: Array<object>) {
+    if (arrayA.length !== arrayB.length) {
+        return false;
+    }
+
+    return arrayA.every((element, index) => {
+        return JSON.stringify(element) === JSON.stringify(arrayB[index]);
```

Comment:
This is potentially expensive, could the comparison just be on a single property (e.g. localAccountId)?

Reply:
Yeah I considered that, do we think the other properties on `AccountInfo` are stable? We know `localAccountId` and `homeAccountId` should never change, but is it possible someone could change their `username` for example? That could potentially leave us in a bad state if we're only updating on the presence or absence of a single property.

Reply:
> We know localAccountId and homeAccountId should never change

I think a check for one or both of these properties should be fine.

Reply:
v1 already solved this actually: https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-core/src/Account.ts#L81

Reply:
@jasonnutter Updated. Thoughts?

Reply:
I think it would be good to have tests, but like it otherwise!

Reply:
Will have to add `idTokenClaims` to the comparison once your PR is merged. That's an example of a potentially unstable value we would want to update state on.